tests/impl.cpp

```diff
@@ -4071,11 +4085,37 @@ result_t test_mm_div_sd(const SSE2NEONTestImpl &impl, uint32_t i)
 result_t test_mm_extract_epi16(const SSE2NEONTestImpl &impl, uint32_t i)
 {
     uint16_t *_a = (uint16_t *) impl.mTestIntPointer1;
-    const int imm = 1;
 
     __m128i a = do_mm_load_ps((const int32_t *) _a);
-    int c = _mm_extract_epi16(a, imm);
-    ASSERT_RETURN(c == *(_a + imm));
+    int c;
+    switch (i & 0x7) {
+        case 0:
+            c = _mm_extract_epi16(a, 0);
+            break;
+        case 1:
+            c = _mm_extract_epi16(a, 1);
+            break;
+        case 2:
+            c = _mm_extract_epi16(a, 2);
+            break;
+        case 3:
+            c = _mm_extract_epi16(a, 3);
+            break;
+        case 4:
+            c = _mm_extract_epi16(a, 4);
+            break;
+        case 5:
+            c = _mm_extract_epi16(a, 5);
+            break;
+        case 6:
+            c = _mm_extract_epi16(a, 6);
+            break;
+        case 7:
+            c = _mm_extract_epi16(a, 7);
+            break;
+    }
+
+    ASSERT_RETURN(c == _a[i&0x7]);
```

Comment:
Ditto.
And you can keep the original code writing format like `ASSERT_RETURN(c == *(_a + idx));`.